### PR TITLE
add doc for perf qps tests with steps for running

### PR DESCRIPTION
**Pull request by apolcyn**
cc @MakMukhi @mehrdada

### CODE REVIEW

File: tools/run_tests/performance/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+Example to count syscalls in grpc-go server during a benchmark:
+* Connect to server machine and run:
+```
+$ netstat -tulpn | grep <driver_port> # to get pid of worker
+$ perf stat -p <worker_pid> -e syscalls:sys_enter_write # stop after test complete
+```
+
+Example memory profile of grpc-go server, with `go tools pprof`:
+* After a run is done on the server, see it's alloc profile with:
```
Comment (mehrdada) [inline tools/run_tests/performance/README.md]: nit: s/it's/its

**Comment by jtattermusch** [comment]
@apolcyn  thanks for writing this, we really need good documentation.

**Comment by adelez** [comment]
+1 to what @jtattermusch said. Fixes #10248

File: tools/run_tests/performance/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+The [run_performance_test.py](../run_performance_tests.py) top-level runner script can also
+be used with remote machines, but for e.g., profiling the server,
+it might be useful to run workers manually.
+
+1. You'll need a "driver" and separate "worker" machines.
+For example, you might use one GCE "driver" machine and 3 other
+GCE "worker" machines that are in the same zone.
+
+2. Connect to each worker machine and start up a benchmark worker with a "driver_port".
+  * For example, to start the grpc-go benchmark worker:
```
Comment (adelez) [inline tools/run_tests/performance/README.md]: Can you please list the commands for running workers in other languages?

Comment (apolcyn) [inline tools/run_tests/performance/README.md]: latest updates add in the different language build commands

**Comment by apolcyn** [comment]
- macos failures:
  * https://github.com/grpc/grpc/issues/10413
  * https://github.com/grpc/grpc/issues/9504
  * there was a mac build failure around c++ benchmarks code but I think https://github.com/grpc/grpc/pull/10401 fixes

- tsan pr: https://github.com/grpc/grpc/issues/9637 (closed currently)
- interop pr: https://github.com/grpc/grpc/issues/9011 (fixed)

**Pull request merged**